Sources/AsyncDNSResolver/dnssd/DNSResolver_dnssd.swift

```diff
@@ -179,6 +176,27 @@ struct DNSSD {
 
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 extension DNSSD {
+    // Class used to manage both the handler and the serviceRef pointers, so that they are deallocated when we are done using them
+    class Query {
+        let serviceRefPtr = UnsafeMutablePointer<DNSServiceRef?>.allocate(capacity: 1)
+
+        let replyHandlerPointer = UnsafeMutableRawPointer.allocate(
+            byteCount: MemoryLayout<QueryReplyHandler>.stride,
+            alignment: MemoryLayout<QueryReplyHandler>.alignment
+        )
```

Comment:
Is there a difference in initializing `serviceRefPtr` and `replyHandlerPointer` here vs. inside the initializer? Does anyone have any preference?

Reply:
Slight preference towards doing it in the `init` so the `allocate` and `initializeMemory` calls are next to each other.